### PR TITLE
 Create integration test suite and migrate appropriate tests from unit test suite #587

### PR DESCRIPTION
**Pull request by erangeles**
Connects to #587

[Review on Reviewable](https://reviewable.io/reviews/softlayer/sl-ember-components/842)

### CODE REVIEW

File: tests/unit/components/sl-drop-option-test.js [hunk 1]
```diff
@@ -6,48 +5,35 @@ moduleForComponent( 'sl-drop-option', 'Unit | Component | sl drop option', {
...
-});
+test( 'Properties have correct default values', function( assert ) {
+    const component = this.subject();
 
-test( 'Has expected aria-role property', function( assert ) {
     assert.strictEqual(
-        this.$().attr( 'role' ),
-        'menuitem',
-        'ARIA role is properly set to "menuitem"'
+        component.get( 'data' ),
```
Comment (notmessenger) [inline tests/unit/components/sl-drop-option-test.js]: This assertion is duplicated.

**Comment by notmessenger** [comment]
Add a unit test for `tagName`

**Comment by notmessenger** [comment]
The function of `optionType()` should be unit tested

File: tests/integration/components/sl-drop-option-test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent( 'sl-drop-option', 'Integration | Component | sl drop option', {
+    integration: true
+});
+
+test( 'Has expected initial class name', function( assert ) {
+    this.render( hbs`
+        {{#sl-drop-option}}
```
Comment (notmessenger) [inline tests/integration/components/sl-drop-option-test.js]: Change template to `{{sl-drop-option}}`

File: tests/integration/components/sl-drop-option-test.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        {{/sl-drop-option}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'sl-drop-option' ),
+        'Rendered component has class "sl-drop-option"'
+    );
+
+    assert.ok(
+        this.$( '>:first-child' ).find( 'li#tagName' ),
```
Comment (notmessenger) [inline tests/integration/components/sl-drop-option-test.js]: What is `#tagName`?

File: tests/integration/components/sl-drop-option-test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        'Rendered componenet has the correct "li" tag'
+    );
+});
+
+test( 'Has expected aria-role property', function( assert ) {
+    this.render( hbs`
+        {{#sl-drop-option}}
```
Comment (notmessenger) [inline tests/integration/components/sl-drop-option-test.js]: Change template to `{{sl-drop-option}}`

File: tests/integration/components/sl-drop-option-test.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    assert.strictEqual(
+        this.$( '>:first-child' ).attr( 'role' ),
+        'menuitem',
+        'ARIA role is properly set to "menuitem"'
+    );
+});
+
+test( 'Option type class value depends on `label` value', function( assert ) {
+    this.render( hbs`
+        {{#sl-drop-option}}
```
Comment (notmessenger) [inline tests/integration/components/sl-drop-option-test.js]: Change template to `{{sl-drop-option}}`

File: tests/integration/components/sl-drop-option-test.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        'Rendered component initially does not have class "presentation"'
+    );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'divider' ),
+        'Rendered component initially has class "divider"'
+    );
+
+    this.render( hbs`
+        {{#sl-drop-option label="test"}}
```
Comment (notmessenger) [inline tests/integration/components/sl-drop-option-test.js]: Change template to `{{sl-drop-option}}`

**Comment by notmessenger** [comment]
Missing integration tests for:

* `label` in template
* `icon` in template
* action in template

**Comment by notmessenger** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)

**Comment by notmessenger** [comment]
Connects to #587

**Comment by erangeles** [comment]
Review status: 0 of 2 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 10 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-10)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L10)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 20 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-20)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L20)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 27 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-27)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L27)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 40 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-40)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L40)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 56 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-56)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L56)):</sup>
Done.

---

<sup>**[tests/unit/components/sl-drop-option-test.js, line 12 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLF-r1-12)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/unit/components/sl-drop-option-test.js#L12)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)

**Comment by erangeles** [comment]
Added a unit test for tagname, function of `optionType()` test added, added integration test for label and icon but action has been left in unit/component. btw I think comments are better placed inline with this tool for readability sakes/acknowledgement. 
 

---

Review status: 0 of 2 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)

**Comment by notmessenger** [comment]
Action needs to be tested in both places.  The reason for the integration test is to make sure that the `action` property is being properly wired up in https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/addon/templates/components/sl-drop-option.hbs#L2 from a template perspective.

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 87 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-K-04c09NUNC2I1Lj7c0)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0c3d75bdd6ebef2330bee67d6f0a88fac8d94f66/tests/integration/components/sl-drop-option-test.js#L87)):</sup>
Should also assert that the `label` property becomes the text of the hyperlink

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 89 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-K-03SIpZieSytPVvHko)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0c3d75bdd6ebef2330bee67d6f0a88fac8d94f66/tests/integration/components/sl-drop-option-test.js#L89)):</sup>
Spelling of "renedred"

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)

**Comment by erangeles** [comment]
Addressed 2/3 comments. Still working on action integration test, not ready to review.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 87 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-K-04c09NUNC2I1Lj7c0)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0c3d75bdd6ebef2330bee67d6f0a88fac8d94f66/tests/integration/components/sl-drop-option-test.js#L87)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 89 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-K-03SIpZieSytPVvHko)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0c3d75bdd6ebef2330bee67d6f0a88fac8d94f66/tests/integration/components/sl-drop-option-test.js#L89)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)

**Comment by notmessenger** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)

**Comment by erangeles** [comment]
implemented action integration test with mcarrol's proposed pattern https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/tests/integration/components/sl-calendar-test.js#L178 . 

---

Review status: 1 of 2 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)

**Comment by notmessenger** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)

**Pull request merged**